src/components/Friendly-captcha/Frontend/useCaptchaHook.tsx

```diff
@@ -36,11 +46,13 @@ function FC_PUZZLE_EP(endpoint: FriendCaptchaEndpoint): string {
 }
 
 const FriendlyCaptcha = (props: FriendlyCaptchaWidgetProps) => {
+  // container is set in return function, where the <div> gets the container as ref attribute.
   const container = useRef<HTMLDivElement | null>(null);
   const widget = useRef<WidgetInstance | null>(null);
 
   useEffect(() => {
-    if (!widget.current && container.current) {
+    if (!props.captchaRendered && container.current) {
+      props.setCaptchaRendered(true);
```

Comment:
Recommend to put this below `new WidgetInstance` as you set the work cleared before clearing the work.